client-v2/src/actions/Collections.ts

```diff
@@ -3,77 +3,97 @@ import { batchActions } from 'redux-batched-actions';
-import { Collection } from 'shared/types/Collection';
+import { frontStages } from 'constants/fronts';
+import { Stages, Collection } from 'shared/types/Collection';
 import { recordUnpublishedChanges } from 'actions/UnpublishedChanges';
 import difference from 'lodash/difference';
 
 function getCollection(collectionId: string): ThunkResult<Promise<string[]>> {
-  return (dispatch: Dispatch, getState: () => State) => {
+  return async (dispatch: Dispatch, getState: () => State) => {
     dispatch(collectionActions.fetchStart(collectionId));
-    return fetchCollection(collectionId)
-      .then(res => {
-        const collectionConfig = getCollectionConfig(getState(), collectionId);
-        const collectionWithNestedArticles = combineCollectionWithConfig(
-          collectionConfig,
-          res
-        );
-        const hasUnpublishedChanges =
-          collectionWithNestedArticles.draft !== undefined;
+    try {
+      const res = await fetchCollection(collectionId)
+      const collectionConfig = getCollectionConfig(getState(), collectionId);
+      const collectionWithNestedArticles = combineCollectionWithConfig(
+        collectionConfig,
+        res
+      );
+      const hasUnpublishedChanges =
+        collectionWithNestedArticles.draft !== undefined;
+
+      const collectionWithDraftArticles = {
+        ...collectionWithNestedArticles,
+        draft: populateDraftArticles(collectionWithNestedArticles)
+      };
+      const {
+        collection,
+        articleFragments,
+        groups
+      } = normaliseCollectionWithNestedArticles(
+        collectionWithDraftArticles,
+        collectionConfig
+      );
+
+      dispatch(
+        batchActions([
+          collectionActions.fetchSuccess(collection),
+          articleFragmentsReceived(articleFragments),
+          recordUnpublishedChanges(collectionId, hasUnpublishedChanges),
+          groupsReceived(groups),
+        ])
+      );
 
-        const collectionWithDraftArticles = {
-          ...collectionWithNestedArticles,
-          draft: populateDraftArticles(collectionWithNestedArticles)
-        };
-        const {
-          collection,
-          articleFragments,
-          groups
-        } = normaliseCollectionWithNestedArticles(
-          collectionWithDraftArticles,
-          collectionConfig
-        );
+      const state = getState();
+      const liveVisibleStories = await getVisibleStories(collection, state, frontStages.live);
+      const draftVisibleStories = await getVisibleStories(collection, state, frontStages.draft);
 
-        dispatch(
-          batchActions([
-            collectionActions.fetchSuccess(collection),
-            articleFragmentsReceived(articleFragments),
-            recordUnpublishedChanges(collectionId, hasUnpublishedChanges),
-            groupsReceived(groups)
-          ])
-        );
+      dispatch(
+        batchActions([
+          recordVisibleStories(collection.id, liveVisibleStories, frontStages.live),
```

Comment:
Is there a way we can create this action with a shape that means we only have to call it once with a subset of either `live` or `draft` in the payload? Maybe that isn't desirable anyway?